engine/termops.ml

```diff
@@ -994,12 +994,14 @@ let rec strip_outer_cast sigma c = match EConstr.kind sigma c with
 (* flattens application lists throwing casts in-between *)
 let collapse_appl sigma c = match EConstr.kind sigma c with
   | App (f,cl) ->
+    if EConstr.isCast sigma f then
```

Comment:
Does this change the behavior when you have nested App nodes and a cast underneath the outermost App?  Is there an invariant that prevents nested App nodes?